amazon-s3/6.0/modules/ROOT/pages/amazon-s3-connector-reference.adoc

```diff
@@ -0,0 +1,3652 @@
+| Object key a| String | Object key for which the multipart upload is initiated. |  | x
+| Upload Id a| String |  Upload ID that identifies the initiated multipart upload. |  | x
+| Completed Parts a| Array of <<MultipartPart>> | List of completed parts. |  `#[payload]` |
+| Config Ref a| ConfigurationProvider |  Name of the configuration to use to execute this component. |  | x
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  `#[payload]` |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+==== Output
+
+[%autowidth.spread]
+|===
+|Type |<<CompleteMultipartUploadResponse>>
+|===
+
+==== Associated Configurations
+
+* <<config>>
+
+==== Throws
+
+* S3:BAD_REQUEST
+* S3:CONNECTIVITY
+* S3:FORBIDDEN
+* S3:NO_SUCH_UPLOAD
+* S3:RETRY_EXHAUSTED
+* S3:S3_ERROR
+* S3:SERVER_BUSY
+* S3:SERVICE_ERROR
+
+
+[[CopyObject]]
+=== Copy Object
+`<s3:copy-object>`
+
+
+Creates a copy of an object that is already stored in Amazon S3.
+
+You can store individual objects of up to 5 TB in Amazon S3. You create a copy of your object up to 5 GB in size in a single atomic action using this API. However, to copy an object greater than 5 GB, you must use the multipart upload *Upload Part - Copy API*. For more information, refer to https://docs.aws.amazon.com/AmazonS3/latest/dev/CopyingObjctsUsingRESTMPUapi.html[Copy Object Using the REST Multipart Upload API].
+
+All copy requests must be authenticated. Additionally, you must have read access to the source object and write access to the destination bucket. For more information, refer to https://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html[REST Authentication].
+
+Both the region that you want to copy the object from and the region that you want to copy the object to must be enabled for your account. A copy request might return an error when Amazon S3 receives the copy request or while Amazon S3 is copying the files. If the error occurs before the copy action starts, you receive a standard Amazon S3 error. If the error occurs during the copy operation, the error response is embedded in the `200` OK response. This means that a `200` OK response can contain either a success or an error.
+
+Design your application to parse the contents of the response and handle it appropriately. If the copy is successful, you receive a response with information about the copied object. If the request is an HTTP 1.1 request, the response is chunk encoded. If it is not, it does not contain the Content-Length, and you would need to read the entire body. The copy request charge is based on the storage class and region that you specify for the destination object. For pricing information, refer to http://aws.amazon.com/s3/pricing/[Amazon S3 pricing].
```

Comment:
```suggestion
Design your application to parse the contents of the response and handle it appropriately. If the copy is successful, you receive a response with information about the copied object. If the request is an HTTP 1.1 request, the response is chunk encoded. Otherwise, it does not contain the Content-Length, and you must read the entire body. The copy request charge is based on the storage class and region that you specify for the destination object. For pricing information, refer to http://aws.amazon.com/s3/pricing/[Amazon S3 pricing].
```